src/assets/scripts/client/base/BaseModel.js

```diff
@@ -45,4 +49,22 @@ export default class BaseModel {
     reset() {
         throw new TypeError('BaseModel#reset method must be implemented by the class extending BaseModel');
     }
+
+    /**
+     * This will verify if the given argument is a string otherwise it will return 'Base Model'
+     *
+     * @for BaseModel
+     * @method veriftyModelName
+     * @param {string / Object}
+     */
+    buildIdPrefix(modelName) {
+        if (!_isString(modelName)) {
+            console.error('BaseModel#constructor expects a string for its first paramert but a string was not given');
+            return 'BaseModel-';
```

Comment:
correctL `BaseModel-1234` will be the end result.  and since `_uniqueId()` takes care of the `1234` part of it, we only need the `BaseModel-` part.